dataprofiler/tests/data_readers/test_filepath_or_buffer.py

```diff
@@ -0,0 +1,123 @@
+import unittest
+import os
+from io import open, StringIO, BytesIO
+
+import pandas as pd
+
+from dataprofiler.data_readers.filepath_or_buffer import FileOrBufferHandler
+
+
+MODULE_PATH = os.path.dirname(os.path.abspath(__file__))
+test_root_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+
+class TestFilepathOrBuffer(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        
+        test_dir = os.path.join(test_root_path, 'data')
+        cls.input_file_names = [
+            dict(path=os.path.join(test_dir, 'csv/diamonds.csv'),
+                 count=53940, delimiter=',', has_header=[0],
+                 num_columns=10, encoding='utf-8'),
+            dict(path=os.path.join(test_dir, 'csv/iris.csv'),
+                 count=150, delimiter=',', has_header=[0],
+                 num_columns=6, encoding='utf-8'),  
+            dict(path=os.path.join(test_dir, 'json/iris-utf-8.json'), encoding='utf-8', count=150),
+            dict(path=os.path.join(test_dir, 'txt/discussion_hn.txt'), count=10000),
+            dict(path=os.path.join(test_dir, 'txt/discussion_reddit.txt'), count=10000),   
+        ]
+        cls.output_file_path = None
+
+    def test_make_buffer_from_filepath(self):
+        """
+        Make sure FileOrBufferHandler can input a file and read it similarly to open()
+        """
+        for input_file in self.input_file_names:
+            with FileOrBufferHandler(input_file['path'], 'r') as filepath_or_buffer, open(input_file['path'], 'r') as input_file_check:
+                # check every 10th line for the first 100 lines
+                for i in range(0, 10):    
+                    for i in range(0, 9):
```

Comment:
i assume the files are small enough that just doing the entire file wouldn't be expensive?